src/typescript/frontend/src/components/pages/emojicoin/components/trade-history/table-row/index.tsx

```diff
@@ -130,7 +130,9 @@ const TableRow = ({ item, showBorder }: TableRowDesktopProps) => {
               " my-auto ml-auto mr-[20px]"
             }
           >
-            {"0x" + truncateAddress(item.swapper).substring(2).toUpperCase()}
+            {item.swapper.startsWith("0x")
+              ? "0x" + truncateAddress(item.swapper).substring(2).toUpperCase()
```

Comment:
```suggestion
              ? prettifyHex(item.swapper)
```

There's a `prettifyHex` function I added to `frontend/src/lib/utils` a few days ago you can use here